osu.Game/Overlays/Mods/EditPresetPopover.cs

```diff
@@ -0,0 +1,203 @@
+                        Label = CommonStrings.Name,
+                        TabbableContentContainer = this
+                    },
+                    descriptionTextBox = new LabelledTextBox
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                        Label = CommonStrings.Description,
+                        TabbableContentContainer = this
+                    },
+                    new OsuScrollContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        Height = 100,
+                        Padding = new MarginPadding(7),
+                        Child = scrollContent = new FillFlowContainer
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Padding = new MarginPadding(7),
+                            Spacing = new Vector2(7),
+                        }
+                    },
+                    new FillFlowContainer
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                        Direction = FillDirection.Horizontal,
+                        AutoSizeAxes = Axes.Y,
+                        Spacing = new Vector2(7),
+                        Children = new Drawable[]
+                        {
+                            useCurrentModButton = new ShearedButton
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Text = "Use Current Mods",
+                                Action = trySaveCurrentMod
+                            },
+                            createButton = new ShearedButton
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Text = Resources.Localisation.Web.CommonStrings.ButtonsSave,
+                                Action = tryEditPreset
+                            },
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Body.BorderThickness = 3;
+            Body.BorderColour = colours.Orange1;
+
+            nameTextBox.Current.Value = preset.Name;
+            descriptionTextBox.Current.Value = preset.Description;
+
+            createButton.DarkerColour = colours.Orange1;
+            createButton.LighterColour = colours.Orange0;
+            createButton.TextColour = colourProvider.Background6;
+
+            selectedMods.BindValueChanged(_ => updateActiveState(), true);
+
+            scrollContent.ChildrenEnumerable = preset.Mods.Select(mod => new ModPresetRow(mod));
```

Comment:
This is super haphazard. The same code exists in two different places, which is generally a bad sign. Please combine all state update logic into one method which at most is called from two places.